foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/databaseaccess/DatabaseCall.java

```diff
@@ -1181,6 +1181,9 @@ public void translate(AbstractRecord translationRow, AbstractRecord modifyRow, A
                         parametersValues.add(value);
                     }
                 } else if (parameterType == LITERAL) {
```

Comment:
Interesting. So the parameterType is DatasourceCall.LITERAL and not DatasourceCall.OUT_CURSOR? That seems wrong, right?

Reply:
Eww, `org.eclipse.persistence.internal.databaseaccess.DatabaseCall.prepareInternalParameters(AbstractSession session)` looks like it sets the type as `DatasourceCall.LITERAL`:
https://github.com/eclipse-ee4j/eclipselink/blob/a842a6e2d9a64c1d26210bdf2179e17942035fd7/foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/databaseaccess/DatabaseCall.java#L788-L789

I don't particularly like the logic of "well, LITERAL doesn't do anything, so lets just set it as a LITERAL type". That thinking becomes void if we start adding logic into LITERAL! Now, we are setting the index during translate if the type is OUT_CURSOR... Maybe we can fix this up?

Reply:
Gross... This might extend to other places too... 
https://github.com/eclipse-ee4j/eclipselink/blob/a842a6e2d9a64c1d26210bdf2179e17942035fd7/jpa/org.eclipse.persistence.jpa/src/main/java/org/eclipse/persistence/internal/jpa/StoredProcedureQueryImpl.java#L438-L447
It might not be entirely safe to undo this gross logic... thoughts?